solver/src/solver.rs

```diff
@@ -33,26 +37,111 @@ mod solver_utils;
 // minus this duration to account for additional delay for example from the network.
 const TIMEOUT_SAFETY_BUFFER: Duration = Duration::from_secs(5);
 
-/// Interface that all solvers must implement
+/// Interface that all solvers must implement.
+///
 /// A `solve` method transforming a collection of `Liquidity` (sources) into a list of
 /// independent `Settlements`. Solvers are free to choose which types `Liquidity` they
-/// would like to include/process (i.e. those already supported here or their own private sources)
-/// The `name` method is included for logging purposes.
+/// would like to process, including their own private sources.
 #[async_trait::async_trait]
 pub trait Solver {
     /// The returned settlements should be independent (for example not reusing the same user
     /// order) so that they can be merged by the driver at its leisure.
     async fn solve(&self, orders: Vec<Liquidity>, gas_price: f64) -> Result<Vec<Settlement>>;
 
-    /// Solver's account that should be used to submit settlements.
+    /// Returns solver's account that should be used to submit settlements.
     fn account(&self) -> &Account;
 
-    /// Displayable name of the solver. Defaults to the type name.
+    /// Returns displayable name of the solver.
+    ///
+    /// This method is used in logging and metrics. By default, returns the type name.
     fn name(&self) -> &'static str {
         std::any::type_name::<Self>()
     }
 }
 
+/// Solver ID can be used to identify a solver within a [`Solvers`] collection.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct SolverId(usize);
```

Comment:
Having an integer ID is cool and all, but I suspect we would like to see account address when logging these things. Perhaps we could implement `Display` for these solvers and include both `name` and `account`. These are the most useful values to have for debugging purposes.

Reply:
Well, we don't really display it anywhere, we convert it to solver name instead. Although your comment gave me an idea how to implement it in a better way.